scripts/generate_psa_wrappers.py

```diff
@@ -10,225 +10,14 @@
 ### for robustness and maintainability.
 
 import argparse
-import itertools
```

Comment:
The new file containing `class BufferParameter` and so on needs to be in the same repository for two reasons: because it'll apply to all branches/repos so it needs to be in the framework repository, and also so that `git diff --color-moved` shows it as a move, not as deletion. I've stopped reviewing at a8d6b102a2b6dbb3e2a71f5dcebc51bda5af6632 because as it stands now, it's just appearing to delete code that needs to stay.